### DISCUSSION TITLE
encodeFunctionData for method with overload params

### DISCUSSION THREAD

**Discussion by mqklin** [wevm/viem#1404] [Question]
I have 2 methods for 1 func name in ABI:
```
const abi = [
  {
    inputs: [
      {
        name: 'signalIDs',
      },
      {
        name: 'salt',
      },
    ],
    name: 'closeSignals',
    type: 'function',
  },
  {
    inputs: [
      {
        name: 'signalIDs',
      },
    ],
    name: 'closeSignals',
    type: 'function',
  },
]
```

How do I call `encodeFunctionData` for the 2nd one? 
```
encodeFunctionData({
          abi,
          functionName: 'closeSignals',
          args: [
            'abc',
          ],
        });
```

Now I'm getting this error: 
```
AbiEncodingLengthMismatchError: ABI encoding params/values length mismatch.
Expected length (params): 2
Given length (values): 1
```

**Comment by jxom** [comment]
That ABI is invalid. The `input`s need a `type`.

**Reply by mqklin** [reply]
Yeah, I cropped the ABI. I simplified code to show the problem.

I found the way btw:
```
...
abi: abi.filter(({name, inputs}) => name === 'closeSignals' && inputs.length === 1),
```
Is it the correct way?

**Reply by jxom** [reply]
Viem should be able to extract the correct ABI item. Can you please provide the full ABI so I can take a look?

**Reply by mqklin** [reply]
https://sanr-l2-api.production.internal.santiment.net/api/v1/abi/0x70fDa011B629AA0Ae6454dd0fF38ee8eF5618c27 (`"name": "closeSignals"`) (Actually I'm trying to call closeSignals with 5 args, but get this error: 
```
AbiEncodingLengthMismatchError: ABI encoding params/values length mismatch.
Expected length (params): 6
Given length (values): 5
```
)
